### PR TITLE
[LPT] optimizeMultiplierAfter refactorinig & Reshape fix

### PR DESCRIPTION
**Pull request by eshoguli**
### Details:
 - [LPT] optimizeMultiplierAfter refactorinig & Reshape fix

### Tickets:
 - *58277*
 - Validation: *62421*

### CODE REVIEW

File: inference-engine/src/low_precision_transformations/src/network_helper.cpp [hunk 1]
```diff
@@ -463,7 +463,17 @@ std::shared_ptr<ngraph::opset1::Multiply> NetworkHelper::optimizeMultipliesAfter
...
-            auto newConst = fold<opset1::Multiply>(constant1, constant2);
+            auto multiplyResult = fold<opset1::Multiply>(constant1, constant2);
+            {
+                // optimize constant shape: used in rfcn-resnet101-coco
+                const auto multiplyResultConstant = as_type_ptr<opset1::Constant>(multiplyResult);
+                if ((multiplyResultConstant != nullptr) && NetworkHelper::isScalarLike(multiplyResultConstant)) {
+                    multiplyResult = std::make_shared<opset1::Constant>(
+                            multiplyResultConstant->get_element_type(),
+                            Shape{},
+                            multiplyResultConstant->cast_vector<float>()[0]);
```
Comment (v-Golubev) [inline inference-engine/src/low_precision_transformations/src/network_helper.cpp:474]: To my mind it is better to use `multyplyResult = NetworkHelper::toScalar(multiplyResultConstant)`

File: inference-engine/src/low_precision_transformations/src/reshape.cpp [hunk 2]
```diff
@@ -47,7 +47,7 @@ void reshapeDequantizationConstant(const std::shared_ptr<opset1::Reshape>& resha
     auto replaceConstant = [](const std::shared_ptr<opset1::Reshape>& reshape, const std::shared_ptr<opset1::Constant>& originalConstant) {
         // reshape for element-wise constant is not required
         auto constantShape = originalConstant->get_shape();
-        if (shape_size(constantShape) == 1ul) {
+        if ((shape_size(constantShape) == 1ul) || (NetworkHelper::isScalarLike(originalConstant))) {
```
Comment (v-Golubev) [inline inference-engine/src/low_precision_transformations/src/reshape.cpp:50]: `if (NetworkHelper::isScalarLike(originalConstant))` (isScalarLike contains check on `shape_size(constantShape == 1ul)`)

**Review by v-Golubev — COMMENTED** [review]
Only minor comments

**Comment by v-Golubev** [comment]
@vzinovie, Could you please review?

**Review by vzinovie — APPROVED** [review]

**Pull request merged**